HarfangLab/harfanglab/get_hostnames_by_ip_action.py

```diff
@@ -0,0 +1,41 @@
+# coding: utf-8
+
+# natives
+import requests
+from sekoia_automation.action import Action
+
+# third parties
+from harfanglab.models import HostnamesResult, HostnameEntry
+
+
+class GetHostnamesByIP(Action):
+    """
+    Action to analyze a HarfangLab job that lists the process
+    """
+
+    def run(self, arguments) -> dict:
+        target_ip = arguments.get("target_ip", "")
+        get_only_last_seen = arguments.get("get_only_last_seen", False)
+
+        instance_url: str = self.module.configuration["url"]
+        api_token: str = self.module.configuration["api_token"]
+
+        job_url = f"{instance_url}/api/data/endpoint/Agent/"
+        params: dict = {"ipaddress": target_ip}
+
+        response = requests.get(url=job_url, json=params, headers={"Authorization": f"Token {api_token}"})
+        response.raise_for_status()
+
+        data = response.json()
+
+        # Check if data is a dictionary with a key "result" or directly a list
+        result = data.get("result") if isinstance(data, dict) else data
```

Comment:
A little typo on the name of the field that contains the items in the API response:
```suggestion
        result = data.get("results") if isinstance(data, dict) else data
```